kernel/src/table_properties/deserialize.rs

```diff
@@ -0,0 +1,321 @@
+        props
+    }
+}
+
+// attempt to parse a key-value pair into a `TableProperties` struct. Returns Some(()) if the key
+// was successfully parsed, and None otherwise.
+fn try_parse(props: &mut TableProperties, k: &str, v: &str) -> Option<()> {
+    match k {
+        "delta.appendOnly" => props.append_only = Some(parse_bool(v)?),
+        "delta.autoOptimize.autoCompact" => props.auto_compact = Some(parse_bool(v)?),
+        "delta.autoOptimize.optimizeWrite" => props.optimize_write = Some(parse_bool(v)?),
+        "delta.checkpointInterval" => props.checkpoint_interval = Some(parse_positive_int(v)?),
+        "delta.checkpoint.writeStatsAsJson" => {
+            props.checkpoint_write_stats_as_json = Some(parse_bool(v)?)
+        }
+        "delta.checkpoint.writeStatsAsStruct" => {
+            props.checkpoint_write_stats_as_struct = Some(parse_bool(v)?)
+        }
+        "delta.columnMapping.mode" => {
+            props.column_mapping_mode = ColumnMappingMode::try_from(v).ok()
+        }
+        "delta.dataSkippingNumIndexedCols" => {
+            props.data_skipping_num_indexed_cols = DataSkippingNumIndexedCols::try_from(v).ok()
+        }
+        "delta.dataSkippingStatsColumns" => {
+            props.data_skipping_stats_columns = Some(parse_column_names(v)?)
+        }
+        "delta.deletedFileRetentionDuration" => {
+            props.deleted_file_retention_duration = Some(parse_interval(v)?)
+        }
+        "delta.enableChangeDataFeed" => props.enable_change_data_feed = Some(parse_bool(v)?),
+        "delta.enableDeletionVectors" => props.enable_deletion_vectors = Some(parse_bool(v)?),
+        "delta.isolationLevel" => props.isolation_level = IsolationLevel::try_from(v).ok(),
+        "delta.logRetentionDuration" => props.log_retention_duration = Some(parse_interval(v)?),
+        "delta.enableExpiredLogCleanup" => props.enable_expired_log_cleanup = Some(parse_bool(v)?),
+        "delta.randomizeFilePrefixes" => props.randomize_file_prefixes = Some(parse_bool(v)?),
+        "delta.randomPrefixLength" => props.random_prefix_length = Some(parse_positive_int(v)?),
+        "delta.setTransactionRetentionDuration" => {
+            props.set_transaction_retention_duration = Some(parse_interval(v)?)
+        }
+        "delta.targetFileSize" => props.target_file_size = Some(parse_positive_int(v)?),
+        "delta.tuneFileSizesForRewrites" => {
+            props.tune_file_sizes_for_rewrites = Some(parse_bool(v)?)
+        }
+        "delta.checkpointPolicy" => props.checkpoint_policy = CheckpointPolicy::try_from(v).ok(),
+        "delta.enableRowTracking" => props.enable_row_tracking = Some(parse_bool(v)?),
+        _ => return None,
+    }
+    Some(())
+}
+
+/// Deserialize a string representing a positive integer into an `Option<u64>`. Returns `Some` if
+/// successfully parses, and `None` otherwise.
+pub(crate) fn parse_positive_int(s: &str) -> Option<i64> {
```

Comment:
For these positive int properties, is it worth having `u64` (or even `NonZero<u64>`) as the "official" type (so it's obvious at a glance), but still parse internally as `i64` for java compat?

Reply:
yea I like the sound of that!

Reply:
changed :)